SwrveSDK/SwrveUtils.brs

```diff
@@ -1,67 +1,173 @@
+		SWRVE_EVENT_TYPE_CURRENCY_GIVEN: "currency_given"
+		SWRVE_EVENT_TYPE_IAP: "iap"
+		SWRVE_EVENT_TYPE_SESSION_START: "session_start"
+		SWRVE_EVENT_FIRST_SESSION_STRING: "Swrve.first_session"
+		SWRVE_EVENT_CAMPAIGNS_DOWNLOADED: "Swrve.Messages.campaigns_downloaded"
+		SWRVE_EVENT_AUTOSHOW_SESSION_START: "Swrve.Messages.showAtSessionStart"
+		SWRVE_EVENTS_STORAGE: userID + "SWRVE_EVENTS_STORAGE"
+		SWRVE_JSON_LOCATION: "pkg:/source/JSONFiles/"
+		SWRVE_ASSETS_LOCATION: "tmp:/swrveAssets/"
+		SWRVE_EQUAL: "eq"
+		SWRVE_NOT_EQUAL: "not"
+		SWRVE_AND: "and"
+		SWRVE_OR: "or"
+		SWRVE_BUTTON_DISMISS: "DISMISS"
+		SWRVE_BUTTON_CUSTOM: "CUSTOM"
+		SWRVE_DEFAULT_DELAY_FIRST_MESSAGE: 150
+		SWRVE_DEFAULT_MAX_SHOWS: 99999
+		SWRVE_DEFAULT_MIN_DELAY: 55
+	}
 end function
 
-'Util function to log ints'
-function SWLogI(msg as Integer)
-	if GetGlobalAA().global.SwrveDebug
-		print "[SwrveSDK] " + StrI(msg)
-	end if
+'gets the global Swrve Node'
+function getSwrveNode() as Object
+	return GetGlobalAA().global.Swrve
 end function
 
-'Util function to log floats'
-function SWLogF(msg as Float)
-	if GetGlobalAA().global.SwrveDebug
-		print "[SwrveSDK] " + Str(msg)
+'lig statements under the VERBOSE label and log level'
+sub SWLogVerbose(param1 as Dynamic, param2 = "nil", param3 = "nil", param4 = "nil", param5 = "nil", param6 = "nil", param7 = "nil", param8 = "nil", param9 = "nil", param10 = "nil")
+	SWLogAny([param1, param2, param3, param4, param5, param6, param7, param8, param9, param10], 5)
+end sub
+
+'lig statements under the DEBUG label and log level'
+sub SWLogDebug(param1 as Dynamic, param2 = "nil", param3 = "nil", param4 = "nil", param5 = "nil", param6 = "nil", param7 = "nil", param8 = "nil", param9 = "nil", param10 = "nil")
+	SWLogAny([param1, param2, param3, param4, param5, param6, param7, param8, param9, param10], 4)
+end sub
+
+'lig statements under the INFO label and log level'
+sub SWLogInfo(param1 as Dynamic, param2 = "nil", param3 = "nil", param4 = "nil", param5 = "nil", param6 = "nil", param7 = "nil", param8 = "nil", param9 = "nil", param10 = "nil")
+	SWLogAny([param1, param2, param3, param4, param5, param6, param7, param8, param9, param10], 3)
+end sub
+
+'lig statements under the WARN label and log level'
+sub SWLogWarn(param1 as Dynamic, param2 = "nil", param3 = "nil", param4 = "nil", param5 = "nil", param6 = "nil", param7 = "nil", param8 = "nil", param9 = "nil", param10 = "nil")
+	SWLogAny([param1, param2, param3, param4, param5, param6, param7, param8, param9, param10], 2)
+end sub
+
+'lig statements under the ERROR label and log level'
+sub SWLogError(param1 as Dynamic, param2 = "nil", param3 = "nil", param4 = "nil", param5 = "nil", param6 = "nil", param7 = "nil", param8 = "nil", param9 = "nil", param10 = "nil")
+	SWLogAny([param1, param2, param3, param4, param5, param6, param7, param8, param9, param10], 1)
+end sub
+
+'lig statements under the INFO label and log level'
+sub SWLogAny(paramArr as Object, level as Integer)
```

Comment:
Logging was all or nothing before this change. This logic was added so developers can more easily digest what the SDK is doing.

Reply:
Super 👍